lib/drizzle/io.ex

```diff
@@ -47,57 +47,25 @@ defmodule Drizzle.IO do
     %{gpio: gpio, currstate: 0}
   end
 
-  def handle_cast({:activate, zone}, state) do
-    IO.puts("handle activate #{zone}")
+  defp do_status_change(zone_name, zonestruct, newstate) do
+    intstate = (newstate && 1 || 0)
```

Comment:
You use this one more time and you will probably do not remember what this does down the line.  You can turn this also into a small function and pattern match on the incoming state like this:

```elixir
defp intstate(state) when state in [false, nil], do: 0
defp intstate(state), do: 1
```
then, you use do_status_change like this:
```elixir
  defp do_status_change(zone_name, zonestruct, newstate) do
    newstate = intstate(newstate)
    DrizzleUiWeb.Endpoint.broadcast @topic, "zone status change", %{zone: zone_name, newstate: newstate}
    :ok = Circuits.GPIO.write(zonestruct.gpio, (!newstate)
    newstate
```
You can use different names per your liking of course.
Is it possible to not mess with 0s and 1s as truthy and falsey values, or is it something the hardware needs?

Reply:
values 0 and 1 are just what's being passed to the C library NIF in https://github.com/elixir-circuits/circuits_gpio/blob/master/lib/gpio.ex#L21 - so yeah this is kind of unavoidable when you're talking to real hardware. I've added the priv function and a small comment.